scripts/migrate-yaml.py

```diff
@@ -96,97 +68,58 @@ def do_the_diff(defaults, overrides):
+
+    new_combined_diff = {}
+
+    for key in overrides.keys():
+        if key not in defaults.keys():
+            new_combined_diff[key] = overrides[key]
+        else:
+            if defaults[key] == overrides[key]:
+                pass
+            else:
+                if 'NEW_DEFAULTS' not in new_combined_diff.keys():
+                    new_combined_diff['NEW_DEFAULTS'] = {}
+                if 'OLD_OVERRIDES' not in new_combined_diff.keys():
+                    new_combined_diff['OLD_OVERRIDES'] = {}
+                new_combined_diff['NEW_DEFAULTS'][key] = defaults[key]
+                new_combined_diff['OLD_OVERRIDES'][key] = overrides[key]
+
+    return new_combined_diff
+
+
+def main():
     """Manage the work of loading the files and printing the output as needed
 
-    :param default_files: A list of files to load data from
     :return: exits 0
     """
     args = parse_args()
-    combined_defaults = {}
-
-    # load the configs
-    for value in default_files:
-        combined_defaults.update(parsed_yaml_from(value))
-    yaml_overrides = parsed_yaml_from(args.overrides)
 
     # get the actual diffs
-    returned_diff = do_the_diff(combined_defaults, yaml_overrides)
-
-    # print the diffs
-    if args.danger:
-        print('# Someone used danger-mode, this was not the best idea...')
-        print('---')
-    if returned_diff.added:
-        if args.danger is False:
-            print('==== added the following ====')
-        print(yaml.safe_dump(dict(returned_diff.added),
-                             default_flow_style=False))
-    if args.info and args.danger is False:
-        if returned_diff.removed:
-            print('\n==== removed the following ====')
-            print('==== you shouldn\'t have to care about this... ====')
-            print(yaml.safe_dump(dict(returned_diff.removed),
-                                 default_flow_style=False))
-    if returned_diff.changed_added:
-        if args.danger is False:
-            print('\n==== changed the following (added) ====')
-        print(yaml.safe_dump(returned_diff.changed_added,
-                             default_flow_style=False))
-    if returned_diff.changed_removed and args.danger is False:
-        print('\n==== changed the following (removed) ====')
-        print(yaml.safe_dump(returned_diff.changed_removed,
-                             default_flow_style=False))
+    returned_combined = do_the_diff(defaults=parsed_yaml_from(args.defaults),
+                                    overrides=parsed_yaml_from(args.overrides))
+
+    if os.path.isfile(args.output_file):
+        raise SystemExit('The file you are creating already exists, '
+                         'please move it, we will not overwrite the file.')
+    with open(args.output_file, 'wb') as fh:
+        fh.write(yaml.safe_dump(returned_combined,
+                                default_flow_style=False,
+                                explicit_start=True))
+        fh.flush()
```

Comment:
You shouldn't need to flush anything. When the file is closed it will be flushed. Did you see something that indicated otherwise?

Also, you're opening the file as bytes. Does `yaml.safe_dump` always return bytes or does it return text?